### DISCUSSION TITLE
[IDEA] Scheduling in bulk

### DISCUSSION THREAD

**Discussion by samhoogantink** [pulsecron/pulse#39] [Ideas]
It would be cool if there was a way to schedule in bulk. Currently I'm sending a couple of hundred queries to the database because there's no such way.

**Example**
```typescript
pulse.bulkSchedule('* * * * *', 'RANDOM_NAME', [ ...arrayWithData ]);
``` 

Hope this can be added soon. Thanks for the"rewrite" of Agenda!

**Comment by code-xhyun** [comment]
@samhoogantink 
Is what you want to create multiple tasks with the same name in bulk?

Have you tried this?
```
pulse.schedule('* * * * *', [
	'RANDOM_NAME',
	'RANDOM_NAME',
	'RANDOM_NAME', 
	//... more 
],{
//...any Payload
});
```

**Reply by samhoogantink** [reply]
Yes, that's correct! Or is there already a way to do this? Currently doing the following:
```typescript
for(const item of items) {
    this.context.pulse.schedule('* * * * *', 'EXAMPLE', item);
}
```

**Reply by code-xhyun** [reply]
Oh, your case requires different data for each one.
I will provide this as soon as possible. @samhoogantink

**Reply by samhoogantink** [reply]
Yup! Thank you!

**Reply by samhoogantink** [reply]
Hi,

Will this be released soon?

Here's a variant I made in the mean time. Not ideal, but it's working:

	Pulse.prototype.bulkSchedule = async (when, name, data, priority = "normal") => {
        const jobs = data.map(item => pulse.create(name, item).schedule(when).priority(priority));
		
        const jobsToAdd = [];
        for(const job of jobs) {
            const props = job.toJSON();
            delete props._id;
            delete props.uniqueQuery;
            delete props.uniqueOpts;

            jobsToAdd.push(props);
        }

        const result = await pulse._collection.insertMany(jobsToAdd);
        const insertedIds = Object.values(result.insertedIds);

        return jobs.map((job, index) => {
            job.attrs._id = insertedIds[index];
            return job;
        }) as any[];
	};
    
	declare module '@pulsecron/pulse' {
	    interface Pulse {
	        bulkSchedule: <T extends JobAttributesData>(when: string|Date, name: string, data: T[], priority?: PulsePriorityKeys) => Promise<Job<T>[]>;
	    }
	}

Thanks again.

**Reply by code-xhyun** [reply]
Sorry, I've been busy with work and couldn't handle it. How about submitting your code as a PR and becoming a contributor? @samhoogantink
